File: Easy/2026-05-14-2784-Check-if-Array-is-Good/solution.java
import java.util.HashMap;

class Solution {
    public boolean isGoodArray(int[] nums) {
        int n = 0;
        for (int num : nums) {
            n = Math.max(n, num);
        }
        
        if (nums.length != n + 1) {
            return false;
        }
        
        HashMap<Integer, Integer> countMap = new HashMap<>();
        for (int num : nums) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }
        
        for (int i = 1; i < n; i++) {
            if (countMap.getOrDefault(i, 0) != 1) {
                return false;
            }
        }
        
        return countMap.getOrDefault(n, 0) == 2;
    }
}

File: Easy/2026-05-14-2784-Check-if-Array-is-Good/solution.js
var isGoodArray = function(nums) {
    const n = Math.max(...nums);
    const count = new Array(n + 1).fill(0);
    
    for (const num of nums) {
        count[num]++;
    }
    
    if (count[n] !== 2) return false;
    
    for (let i = 1; i < n; i++) {
        if (count[i] !== 1) return false;
    }
    
    return true;
};